tendrl/gluster_integration/sds_sync/utilization.py

```diff
@@ -40,27 +40,52 @@ def sync_utilization_details(volumes):
     out = ''
     try:
         out, err = cmd.communicate()
+
         util_det = json.loads(out)
-        for k, v in util_det.iteritems():
-            volume = voldict[k]
-            volume.usable_capacity = int(v['total'])
-            volume.used_capacity = int(v['used'])
-            volume.pcnt_used = str(v['pcnt_used'])
-            volume.total_inode_capacity = int(v['total_inode'])
-            volume.used_inode_capacity = int(v['used_inode'])
-            volume.pcnt_inode_used = str(v['pcnt_inode_used'])
-            volume.save()
-            cluster_used_capacity += volume.used_capacity
-            cluster_usable_capacity += volume.usable_capacity
-    except(KeyError, TypeError, ValueError, AttributeError):
+        if err != '':
+            logger.log(
+                "error",
+                NS.publisher_id,
+                {'message' : 'Error : %s\n' %err}
+            )
+        if bool(util_det):
+            if util_det['error']:
```

Comment:
If the key `error` not there it will throw exception so iys better to check if the key exists using `util_det.has_key`

Reply:
I have defined ret_dict to always have 'error' and 'success' as a key, which gets parsed to json as util_det. So, I think it will always have that key?